src/main/resources/hudson/plugins/groovy/WithGroovyStep/help-input.html

```diff
@@ -0,0 +1,14 @@
+<p>
+    Allows the Pipeline script to pass structured objects to the external Groovy script and receive structured return values.
+    The values must be <code>Serializable</code> and may only refer to types defined in the Java Platform or Groovy language.
+    If set to any non-null value, the external script may call the following methods:
```

Comment:
This is sounds like one of the features that turns into a maintenance nightmare after a year or two. -0

Reply:
There's a reason for it -- it's a way to offload processing from the master, so you don't need to do complex logic internally where it'll hit the master itself and can introduce resource use or unsafe-code issues.

Reply:
Right, the key purpose of this PR is to allow us to do this kind of stuff externally rather than inside the master JVM, which is _already_ a maintenance nightmare.

Reply:
What I meant was the input/output variable passing specifically that can be replaced by passing in/out text streams.

Reply:
I think this is the most convenient option for migrating existing scripts which do tricky computation on data structures. As of JEP-200 in 2.102, the allowed types will be those which are whitelisted. Certainly this is open to discussion.